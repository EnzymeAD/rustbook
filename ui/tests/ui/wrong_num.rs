#![feature(autodiff)]
use std::autodiff::autodiff;

#[autodiff(d_square, Reverse, Duplicated, Const, Active)]
fn square(x: &f64) -> f64 {
    x * x
}
