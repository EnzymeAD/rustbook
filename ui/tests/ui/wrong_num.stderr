error: expected 2 activities, but found 3
 --> tests/ui/wrong_num.rs:4:1
  |
4 | #[autodiff(d_square, Reverse, Duplicated, Const, Active)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `autodiff` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/wrong_num.rs:7:2
  |
7 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/wrong_num.rs`

error[E0061]: this function takes 1 argument but 0 arguments were supplied
 --> tests/ui/wrong_num.rs:4:1
  |
4 | #[autodiff(d_square, Reverse, Duplicated, Const, Active)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument #1 of type `&f64` is missing
  |
note: function defined here
 --> tests/ui/wrong_num.rs:5:4
  |
5 | fn square(x: &f64) -> f64 {
  |    ^^^^^^ -------
  = note: this error originates in the attribute macro `autodiff` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
  |
4 | #[autodiff(d_square, Reverse, Duplicated, Const, Active)](/* &f64 */)
  |                                                          ++++++++++++
